pkg/scheduler/controller/shoot/add.go

```diff
@@ -77,3 +79,14 @@ func (r *Reconciler) ShootSpecChangedPredicate() predicate.Predicate {
 		},
 	}
 }
+
+// ShootIsNotAutonomous is a predicate that returns true if a shoot is not mark as 'autonomous shoot cluster' (meaning,
+// that the control plane components run in the same cluster, hence, no seed selection is required).
+func (r *Reconciler) ShootIsNotAutonomous() predicate.Predicate {
+	return predicate.NewPredicateFuncs(func(obj client.Object) bool {
+		if shoot, ok := obj.(*gardencorev1beta1.Shoot); ok {
+			return !helper.IsShootAutonomous(shoot)
+		}
+		return true
```

Comment:
In `ShootUnassignedPredicate()` above the default case is `return false`. Here it is the other way round. We should not call this predicate for anything that is no `Shoot`, but we should be consistent.
WDYT?